src/Vote.Monitor.Api.Feature.Observer/Services/CsvParser.cs

```diff
@@ -0,0 +1,137 @@
+﻿using System.Globalization;
+using CsvHelper;
+using CsvHelper.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Vote.Monitor.Api.Feature.Observer;
+using Vote.Monitor.Api.Feature.Observer.Services;
+
+namespace Vote.Monitor.Api.Feature.PollingStation.Services;
+
+
+
+public class CsvParser<T, TMapper> : ICsvParser<T> where T : class, IDuplicateCheck where TMapper : ClassMap<T>
+{
+    private readonly Validator<T> _modelValidator;
+    private readonly ILogger _logger;
+    private readonly CsvFileConfig _parserConfig;
+
+    public CsvParser(
+        ILogger<CsvParser<T, TMapper>> logger,
+        IOptions<CsvFileConfig> options,
+        Validator<T> modelValidator
+        )
+    {
+
+        _logger = logger;
+        _parserConfig = options.Value;
+        _modelValidator = modelValidator;
+    }
+
+    public ParsingResult2<T> Parse(Stream stream)
+    {
+        try
+        {
+            int numberOfInvalidRows = 0;
+            int rowIndex = 1;
+            var validationErrors = new List<ValidationResult>();
+            List<CsvRowParsed<T>> rowsRead = new List<CsvRowParsed<T>>();
+
+            using var reader = new StreamReader(stream);
+
+            var readerConfiguration = new CsvConfiguration(CultureInfo.InvariantCulture)
+            {
+                ExceptionMessagesContainRawData = false,
+                MissingFieldFound = null
+            };
+
+
+
+            using var csv = new CsvHelper.CsvReader(reader, readerConfiguration);
+            csv.Context.RegisterClassMap<TMapper>();
+
+            while (csv.Read())
+            {
+                T? item = csv.GetRecord<T>();
+                var currentRow = new CsvRowParsed<T>()
+                {
+                    IsSuccess = true,
+                    OriginalRow = csv.Parser.RawRecord,
+                    Value = item
+                };
+                rowsRead.Add(currentRow);
+
+
+                if (item == null)
+                {
+                    ++numberOfInvalidRows;
+                    currentRow.IsSuccess = false;
+                    currentRow.ErrorMessage = $"Malformed row {rowIndex}";
+                    validationErrors.Add(new ValidationResult(new List<ValidationFailure> { new("Csv File", $"Malformed row {rowIndex}") }));
```

Comment:
the use of `validationErrors` is redundant since we keep the validation of the row in `CsvRowParsed`.
Also, we only use it for returning a message to the caller.
Lets just return a generic message `ParsingResult<T>.Fail("Validation failed see result file")

Reply:
removed the errors list.